policybot/pkg/check/check.go

```diff
@@ -0,0 +1,323 @@
+ */
+func contains(slic []string, ele string) bool {
+	for _, e := range(slic) {
+		if strings.Compare(e, ele) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+/*
+ * GetTest function get all directories under the given pr in istio-prow/pr-logs/pull/istio-istio/PRNUMBER for each test suite name.
+ * @param client client used to get buckets and objects.
+ * @param prNum the PR number inputted.
+ * @return []Tests return a slice of Tests objects.
+ */
+func getTests(client *storage.Client, prNum string) []Tests {
+	ctx := context.Background()
+	bucket := client.Bucket("istio-prow")
+	query := &storage.Query{Prefix:    "pr-logs/pull/istio_istio/" + prNum}
+	it := bucket.Objects(ctx, query)
+	var testSlice []Tests
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
+		}
+		name := attrs.Name
+
+		nameSlice := strings.Split(name, "/")
+		prNum := nameSlice[3]
+		pullNum := nameSlice[5]
+		testName := nameSlice[len(nameSlice) - 3]
+		fileName := nameSlice[len(nameSlice)-1] // C
+		var newString = "pr-logs/pull/istio_istio/" + prNum + "/" + testName + "/" + pullNum
+		if strings.Compare(fileName, "started.json") == 0 || strings.Compare(fileName, "clone-records.json") == 0 || strings.Compare(fileName, "finished.json") == 0 {
+			var contain = false
+			for ind, ele := range(testSlice) {
+				if strings.Compare(ele.Name, testName) == 0 {
+					prs := ele.Prs
+
+					if contains(prs, newString) == false {
+						prs = append(prs, newString)
+						ele.Prs = prs
+						testSlice[ind] = ele
+					}
+					
+					contain = true
+				}
+
+			}
+			if contain == false {
+				t := Tests{
+					Name: testName,
+				}
+				newSlice := []string{newString}
+				t.Prs = newSlice
+				testSlice = append(testSlice, t)
+			}
+		}
+	}
+	return testSlice
+}
+
+/*
+ * GetShaAndPassStatus function return the status of test passing, clone failure, sha number, base sha for each test run under each test suite for the given pr.
+ * @param client client used to get buckets and objects from google cloud storage.
+ * @param testSlice a slice of Tests objects containing all tests and the path to folder for each test run for the test under such pr.
+ * @return a map of test suite name -- pr number -- run number -- ForEachRun objects.
+ */
+func getShaAndPassStatus(client *storage.Client, testSlice []Tests) []TestFlake {
+	ctx := context.Background()
+	bucket := client.Bucket("istio-prow")
+
+	var allTestRuns = []TestFlake{}
+
+	for _, test := range(testSlice) {
+		testName := test.Name
+
+		prefs := test.Prs
+
+		for _, pref := range(prefs) {
+
+			var onePull = TestFlake{}
+
+			onePull.TestName = testName
+			onePull.RunPath = pref
+
+			obj := bucket.Object(pref + "/clone-records.json")
+			log.Println("read clone")
+			rdr, err := obj.NewReader(ctx)
+			if err != nil {
+		        log.Println("readFile: unable to open file from bucket %q, file %q: %v", err)
```

Comment:
In general, think of this code as a library someone calls, and not as a stand-alone program. So you shoudn't be logging errors like this, you should be returning errors to the caller and let the caller deal with printigin.